### DISCUSSION TITLE
Translate integer to errno constant?

### DISCUSSION THREAD

**Discussion by daandemeyer** [bpftrace/bpftrace#2082] [Q&A] [answered]
I'd like to translate integers to their corresponding constant name from errno.h (as a string). For example, -1 => "EINVAL". Is this possible with a printf specifier by any chance? I checked the reference guide but it doesn't seem to list the supported specifiers.

**Comment by mmisono** [comment, marked as answer]
There is no printf specifier for that, but it is possible to use a map to convert numbers to strings like [tools/capable.bt](https://github.com/iovisor/bpftrace/blob/master/tools/capable.bt).

**Reply by daandemeyer** [reply]
Would bpftrace itself be interested in maintaining such a map and providing a function to translate an integer to it's errno constant string? I can maintain such a list myself but having to copy it around for each and every script seems a bit inefficient.

**Reply by danobi** [reply]
Yeah I think that's pretty reasonable. Could make a builtin function that does the translation. You'd probably want to store the `int -> string` mapping as a readonly bpf map that the prog can access. And only load the map if the script uses that builtin.

**Reply by viktormalik** [reply]
I created the issue #2114 to track this feature

**Reply by daandemeyer** [reply]
Apologies for not replying, I found https://github.com/anakryiko/retsnoop which fits my use case better rather than trying to implement something similar with bpftrace. I still think having such a mapping in bpftrace itself would be useful to allow for better messages when printing errors.

**Reply by viktormalik** [reply]
Agreed, marking this as an answer and we can continue further discussion on the issue, if needed.
